test/integration/test_components.py

```diff
@@ -149,3 +149,39 @@ def package_info(self):
     # This used to crash, because override was not correctly excluded
     c.run("create app")
     assert "app/0.1: Created package" in c.out
+
+def test_duplication_component_properties():
+    """ Regression for PR 17503 - component lists would be incorrectly aggregated """
+    tc = TestClient(light=True)
+    tc.save({"dep/conanfile.py": textwrap.dedent("""
```

Comment:
I personally think that adding the conanfiles to the save directly with textwrap.dedent is a bit confusing when it comes to review it, I would prefer in general to have them defined in two separate variables, I think it's more clear.